src/mpid/ch4/generic/am/mpidig_am.h

```diff
@@ -77,6 +77,10 @@ typedef int (*MPIDIG_am_target_msg_cb) (int handler_id, void *am_hdr,
 typedef struct MPIDIG_global_t {
     MPIDIG_am_target_msg_cb target_msg_cbs[MPIDI_AM_HANDLERS_MAX];
     MPIDIG_am_origin_cb origin_cbs[MPIDI_AM_HANDLERS_MAX];
+    /* control parameters for global progress of RMA receive active messages */
+    MPL_atomic_int_t rma_recv_am_active;        /* Indicates whether any incoming RMA AM has been received.
+                                                 * Set inside each target callback.*/
```

Comment:
Have the same concern. Same for `rma_recv_am_poll_interval`. But semantically I don't think it makes sense to set recv_am progress per window. Because everytime we make progress for it, it actually handles am globally. So in my opinion. recv_am is always global. We have discussed this yesterday.

We can study the potential performance loss as next step. If we find critical issue, we can fix from there.

Reply:
Will add a node in code for the performance concern.